app/models/miq_server.rb

```diff
@@ -608,4 +608,8 @@ def server_timezone
   def tenant_identity
     User.super_admin
   end
+
+  def objects_in_settings_chain
+    zone ? zone.objects_in_settings_chain << self : [self]
```

Comment:
State of object(s) is not changing since result is not saved anywhere. Calling this method multiple times on the same set of objects(on the same same hierarchy) will produce the same result